### DISCUSSION TITLE
Give Zeek the ability to output the git commit that it was compiled from

### DISCUSSION THREAD

**Discussion by 0xxon** [zeek/zeek#1542] [Ideas]
I recently started to collect performance statistics about a running Zeek installation, which are automatically collected during runtime.

During that time I realized that it would be neat if one could get the exact information of which git commit a running Zeek was compiled from. Having a flag in the zeek executable that outputs this information would make it much easier to add this information to the statistics in an automated way.

I realize that zeek already has the ability to output its version - but especially when you try development branches on a system that is not really sufficient.

From what I understand, Spicy has this ability - so perhaps we could steal the code that does this from there.

**Comment by rsmmr** [comment]
Spicy has this script to extract the version information from git: https://github.com/zeek/spicy/blob/main/scripts/autogen-version (which just got ported from bash to Python for platform portability). The logic has become a bit messy, but could be cleaned up.

**Comment by timwoj** [comment]
You could probably take care of https://github.com/zeek/zeek/issues/1405 as well here.

**Reply by 0xxon** [reply]
Oh, I completely missed that we already have an issue for this.

**Reply by timwoj** [reply]
The issue was more for storing it in a file and just having it in the installation directory. We had some cases with Aashish's testing where we simply didn't know what commit he was on because he was bouncing around between different versions, and having that information would have been really useful.
